### PR TITLE
update envoy vendor

### PR DESCRIPTION
**Pull request by GregHanson**

### CODE REVIEW

**Comment by GregHanson** [comment]
removing the deprecated field in listener.go is currently resulting in the sidecar proxy failing with the following error:
```
[2018-12-28 15:21:13.914][000013][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_mux_subscription_lib/common/config/grpc_mux_subscription_impl.h:70] gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error adding/updating listener 172.17.0.20_9080: cannot bind '172.17.0.20:9080': Address already in use
[2018-12-28 15:21:13.916][000020][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:125] Caught Segmentation fault, suspect faulting address 0x80
[2018-12-28 15:21:13.916][000020][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:94] Backtrace thr<20> obj</usr/local/bin/envoy> (If unsymbolized, use tools/stack_decode.py):
2018-12-28T15:21:13.920403Z    warn    Epoch 0 terminated with an error: signal: segmentation fault (core dumped)
```
From the docs/proto removed in envoy source:
```
-// [#not-implemented-hide:]
-type Listener_DeprecatedV1 struct {
-       // Whether the listener should bind to the port. A listener that doesn’t
-       // bind can only receive connections redirected from other listeners that
-       // set use_original_dst parameter to true. Default is true.
-       //
-       // [V2-API-DIFF] This is deprecated in v2, all Listeners will bind to their
-       // port. An additional filter chain must be created for every original
-       // destination port this listener may redirect to in v2, with the original
-       // port specified in the FilterChainMatch destination_port field.
-       BindToPort           *types.BoolValue `protobuf:"bytes,1,opt,name=bind_to_port,json=bindToPort" json:"bind_to_port,omitempty"`
```

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -1024,19 +1024,11 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 		})
 	}
 
-	var deprecatedV1 *xdsapi.Listener_DeprecatedV1
-	if !opts.bindToPort {
-		deprecatedV1 = &xdsapi.Listener_DeprecatedV1{
-			BindToPort: proto.BoolFalse,
-		}
-	}
-
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: ? why are you removing this?

Comment (GregHanson) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: the DeprecatedV1 field was deleted in envoy:
vendor/github.com/envoyproxy/go-control-plane/envoy/api/v2/listener/listener.pb.go

**Comment by GregHanson** [comment]
closing in favor of https://github.com/istio/istio/pull/10737

**Pull request closed without merging**